homeassistant/components/template/binary_sensor.py

```diff
@@ -54,96 +64,151 @@
     {vol.Required(CONF_SENSORS): cv.schema_with_slug_keys(SENSOR_SCHEMA)}
 )
 
+STORAGE_SCHEMA = SENSOR_SCHEMA.extend({vol.Required(CONF_ID): cv.string})
+
+CREATE_FIELDS = {
+    vol.Required(CONF_FRIENDLY_NAME): vol.All(str, vol.Length(min=1)),
```

Comment:
Any reason we would allow friendly name to be min=1 but for YAML be just string?

Reply:
For the storage collection the entity id is based off of the friendly name, whereas for yaml this is defined separately.